### ISSUE TITLE
[Bug]: express server can't start in production because `remix setup node` didn't run

### ISSUE THREAD

**Issue by smhg** [remix-run/remix#1056]
### Which Remix packages are impacted?

- [ ] `remix` (Remix core)
- [ ] `create-remix`
- [ ] `@remix-run/architect`
- [ ] `@remix-run/cloudflare-workers`
- [ ] `@remix-run/dev`
- [X] `@remix-run/express`
- [ ] `@remix-run/netlify`
- [ ] `@remix-run/node`
- [ ] `@remix-run/react`
- [ ] `@remix-run/serve`
- [ ] `@remix-run/server-runtime`
- [ ] `@remix-run/vercel`

### What version of Remix are you using?

1.0.6

### What version of Node are you using? Minimum supported version is 14.

16.13.1

### Steps to Reproduce

```bash
npx create-remix@latest
// choose express, javascript, run npm install
cd [app-name]
npm run postinstall
npm run build

// 'simulate' production environment:
rm -rf node_modules
npm i --production

npm run start
```

### Expected Behavior

It should be able to start the express server.

### Actual Behavior

```bash
Error: Cannot find module './client'
Require stack:
- my-remix-express-app/node_modules/remix/index.js
- my-remix-express-app/server/build/index.js
- my-remix-express-app/server/index.js
```

This can be solved with `npm install @remix-run/dev` and `remix setup node` (which was the case before `rm -rf node_modules` and `npm i --production` happened).

**Comment by kiliman** [comment]
What Node environment are you running? I use Heroku for simple test projects and it works fine in "production" mode.

Here's my deployment log
```log
-----> Installing dependencies
       Installing node modules
       
       > remix-component-data@0.2.0 postinstall
       > remix setup node
       
       Successfully setup Remix for node.
       ...
-----> Build
       Running build
       
       > remix-component-data@0.2.0 build
       > remix build
       
       Building Remix app in production mode...
       Built in 406ms
       
-----> Caching build
       - node_modules
       
-----> Pruning devDependencies
       
       up to date, audited 96 packages in 1s
       
       found 0 vulnerabilities
       
-----> Build succeeded!
...
-----> Launching...
       Released v8
       https://remix-component-data.herokuapp.com/ deployed to Heroku
```
I'm not sure what exactly Heroku means by *Pruning devDependencies* but I don't think they are nuking node_modules and reinstalling as --production.

**Comment by jacob-ebey** [comment]
`remix setup node` is required. Its job is to setup the "remix" node module to allow importing from there. To get around this in environments that don't install development dependencies *then* purge is to add this to the server start command: `npx @remix-run/dev setup node`. I have ideas to make this no longer required, but for now this is your best option.

**Comment by ryanflorence** [comment]
fixed by #1026
